cmd/controller-manager/main.go

```diff
@@ -161,8 +161,20 @@ func main() {
 		options = append(options, informers.WithNamespace(ns))
 		kubeoptions = append(kubeoptions, kubeinformers.WithNamespace(ns))
 	}
+
+	tweakListOptionsFunc := func(options *metav1.ListOptions) {
+		if len(options.LabelSelector) > 0 {
+			options.LabelSelector += ",app.kubernetes.io/managed-by=tidb-operator"
+		} else {
+			options.LabelSelector += "app.kubernetes.io/managed-by=tidb-operator"
+		}
+	}
+
+	labelKubeOptions := append(kubeoptions, kubeinformers.WithTweakListOptions(tweakListOptionsFunc))
+
 	informerFactory = informers.NewSharedInformerFactoryWithOptions(cli, controller.ResyncDuration, options...)
 	kubeInformerFactory = kubeinformers.NewSharedInformerFactoryWithOptions(kubeCli, controller.ResyncDuration, kubeoptions...)
+	labelFlterKubeInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(kubeCli, controller.ResyncDuration, labelKubeOptions...)
```

Comment:
Em, I think this PR should be merged before https://github.com/pingcap/tidb-operator/pull/3306 and let me resolve the conflicts.